dart/optimizer/pagmo/PagmoUtils.hpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2011-2018, The DART development contributors
+ * All rights reserved.
+ *
+ * The list of contributors can be found at:
+ *   https://github.com/dartsim/dart/blob/master/LICENSE
+ *
+ * This file is provided under the following "BSD-style" License:
+ *   Redistribution and use in source and binary forms, with or
+ *   without modification, are permitted provided that the following
+ *   conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef DART_OPTIMIZER_PAGMO_PAGMOUTILS_HPP_
+#define DART_OPTIMIZER_PAGMO_PAGMOUTILS_HPP_
+
+#include <pagmo/pagmo.hpp>
+#include "dart/optimizer/MultiObjectiveSolver.hpp"
+
+namespace dart {
+namespace optimizer {
+
+class PagmoTypes
+{
+public:
+  static std::vector<double> convertVector(const Eigen::VectorXd& v);
+
+  static Eigen::Map<const Eigen::VectorXd> convertVector(
+      const std::vector<double>& v);
+
+  static Population convertPopulation(
+      const ::pagmo::population& pagmoPop,
+      std::shared_ptr<MultiObjectiveProblem> problem);
```

Comment:
Nitpick: use a `const std::shared_ptr<T>&` here?

Reply:
I intentionally left this as passing by value in order to indicate that this function sharing the ownership because the caller can notice that the reference counting would be increase by the copy. Note that the reference counting would happen only one time because we move the copied `shared_ptr` in the function.

As a note, we would be good to have a documentation on this convention like [this](https://github.com/personalrobotics/aikido/blob/master/STYLE.md#smart-pointers).

Reply:
Ah, okay. That's a subtle but reasonable signal to the user :+1: